### PR TITLE
Create global status flags close #1980

### LINKED ISSUE #1980: Add global flags indicating whether initial queries are completed.
We need to create some flags to indicate whether the initial data for all the entities has been fetched. This is to avoid errors in `clieny.readQuery` when the data is not there. For example when you try to retrieve prices when tickets are loaded, prices may not be there yet.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #1980 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/containers/queries/useFetchTickets.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	const toaster = useToaster();
+	const toasterMessage = 'initializing tickets';
+
+	const { data, error, loading } = useQuery(GET_TICKETS, {
+		variables: {
+			where: {
+				datetimeIn,
+			},
+		},
+		skip: !datetimeIn.length, // do not fetch if we don't have any datetimes
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/queries/useFetchTickets.js]: nice :+1:

File: assets/prototype/domain/eventEditor/index.js [hunk 2]
```diff
@@ -18,15 +19,16 @@ editor.className = 'ee-editor-div-prototype';
 
 container.prepend(editor);
 
-// then load the editor data we dumped into DOM and render theEventEditor
-const eventData = window.eeEditorEventData;
-const { eventId } = eventData;
+const Editor = () => {
+	const eventId = useEventId();
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/index.js]: :+1:

**Comment by nicolad** [comment]
@manzoorwanijk it looks good.
Can you please add an usage based on:
`This is to avoid errors in clieny.readQuery when the data is not there.` from the related issue: https://github.com/eventespresso/event-espresso-core/issues/1980#issue-534290170

**Comment by manzoorwanijk** [comment]
@nicolad That usage will be in #1968 inside `usePriceTypes`

**Pull request merged**